Assets/MixedRealityToolkit/_Core/Utilities/Lines/DataProviders/BaseMixedRealityLineDataProvider.cs

```diff
@@ -464,6 +464,74 @@ public void SetPoint(int pointIndex, Vector3 point)
             SetPointInternal(pointIndex, LineTransform.InverseTransformPoint(point));
         }
 
+        /// <summary>
+        /// Iterates along line until it finds the point closest to worldPosition
+        /// </summary>
+        /// <param name="worldPosition"></param>
+        /// <param name="resolution"></param>
+        /// <param name="maxIterations"></param>
+        /// <returns></returns>
+        public Vector3 GetClosestPoint(Vector3 worldPosition, int resolution = 5, int maxIterations = 5)
+        {
+            float length = GetNormalizedLengthFromWorldPos(worldPosition, resolution, maxIterations);
+            return GetPoint(length);
+        }
+
+        /// <summary>
+        /// Iterates along line until it finds the length closest to worldposition.
+        /// </summary>
+        /// <param name="worldPosition"></param>
+        /// <param name="resolution"></param>
+        /// <param name="maxIterations"></param>
+        /// <returns></returns>
+        public float GetNormalizedLengthFromWorldPos(Vector3 worldPosition, int resolution = 5, int maxIterations = 5)
+        {
+            int iteration = 0;
+            float length = GetNormalizedLengthFromWorldPosInternal(worldPosition, 0f, ref iteration, resolution, maxIterations, 0f, 1f);
+            return length;
+        }
+
+        private float GetNormalizedLengthFromWorldPosInternal(Vector3 worldPosition, float currentLength, ref int iteration, int resolution, int maxIterations, float start, float end)
+        {
+            // If we've maxed out our iterations, don't go any further
+            iteration++;
+
+            if (iteration > maxIterations)
+            {
+                return currentLength;
+            }
+
+            float searchLength;
+            if (start > 0)
+            {
+                searchLength = (end - start) / resolution;
+            }
+            else
+            {
+                searchLength = end / resolution;
+            }
+
+            Vector3 currentPoint = Vector3.zero;
```

Comment:
I'm not sure you need this. You'll be overwriting it each time you get the unclamped point.

Reply:
If you mean the initialization is unnecessary, you're right. It's something I do out of habit. If you mean I could do away with `currentPoint` altogether that's also true, though the result would be less readable.

Reply:
I'm a huge fan of readability. Since you do not read currentPoint before you write to it, you can safely do away with the initialization, but I _do_ like the habit of always initializing variables.

Reply:
I disagree. Harder to immediately understand what is going on.